### PR TITLE
Added Data

### PR DESCRIPTION
**Pull request by mike-ferguson**
Adds data collection plugin as part of the Ferguson2024 Search Asymmetry Benchmark. 

Main Benchmark Plugin PR here #618 
Metric Plugin PR here #882

### CODE REVIEW

File: brainscore_vision/__init__.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def load_ceiling(identifier: str, *args, **kwargs) -> Ceiling:
 def load_benchmark(identifier: str) -> Benchmark:
     import_plugin('brainscore_vision', 'benchmarks', identifier)
 
-    return benchmark_registry[identifier]()
+    return benchmark_registry[identifier]
```
Comment (mschrimpf) [inline brainscore_vision/__init__.py:61]: undo this change

File: brainscore_vision/benchmarks/ferguson2024/__init__.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
+from brainio.assemblies import BehavioralAssembly
+from brainscore_vision import data_registry, stimulus_set_registry, load_stimulus_set
+from brainscore_vision.data_helpers.s3 import load_assembly_from_s3, load_stimulus_set_from_s3
+
+BIBTEX = """TBD"""
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/__init__.py:5]:
```suggestion
BIBTEX = """"""
```

Comment (mschrimpf) [reply brainscore_vision/benchmarks/ferguson2024/__init__.py:5]: leave empty until we put on OSF or so

File: brainscore_vision/benchmarks/ferguson2024/test.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        'gray_easy', 'gray_hard', 'half', 'juncture',
+        'lle', 'llh', 'quarter', 'round_f',
+        'round_v', 'tilted_line'
+    ])
+    def test_distinct_values(self, identifier):
+        assembly = load_dataset(f"Ferguson2024_{identifier}")
+        assert set(assembly['block'].values) == {"first", "second"}
+        assert set(assembly['keypress_response'].values) == {"f", "j"}
+        assert set(assembly['trial_type'].values) == {"normal"}
+        assert set(assembly['distractor_nums'].values) == {"1.0", "5.0", "11.0"}
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/test.py:71]: should these not be float?

File: brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+    assembly = BehavioralAssembly(all_subjects['correct'],
+                                  coords={
+                                      'stimulus_id': ('presentation', all_subjects['stimulus_id']),
+                                      'stimulus_id_long': ('presentation', all_subjects['stimulus']),
+                                      'participant_id': ('presentation', all_subjects['participant_id']),
+                                      'response_time_ms': ('presentation', all_subjects['response_time_ms']),
+                                      'correct': ('presentation', all_subjects['correct']),
+                                      'target_present': ('presentation', all_subjects['target_present']),
+                                      'distractor_nums': ('presentation', all_subjects['distractor_nums']),
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py:76]: floats?

File: brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    all_subjects['stimulus_id'] = all_subjects['stimulus'].apply(extract_and_concatenate)
+
+    assembly = BehavioralAssembly(all_subjects['correct'],
+                                  coords={
+                                      'stimulus_id': ('presentation', all_subjects['stimulus_id']),
+                                      'stimulus_id_long': ('presentation', all_subjects['stimulus']),
+                                      'participant_id': ('presentation', all_subjects['participant_id']),
+                                      'response_time_ms': ('presentation', all_subjects['response_time_ms']),
+                                      'correct': ('presentation', all_subjects['correct']),
+                                      'target_present': ('presentation', all_subjects['target_present']),
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py:75]: bool?

File: brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+    # create an ID that is equal to the stimulus_set ID
+    all_subjects['stimulus_id'] = all_subjects['stimulus'].apply(extract_and_concatenate)
+
+    assembly = BehavioralAssembly(all_subjects['correct'],
+                                  coords={
+                                      'stimulus_id': ('presentation', all_subjects['stimulus_id']),
+                                      'stimulus_id_long': ('presentation', all_subjects['stimulus']),
+                                      'participant_id': ('presentation', all_subjects['participant_id']),
+                                      'response_time_ms': ('presentation', all_subjects['response_time_ms']),
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py:73]: float?

File: brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+                                      'correct': ('presentation', all_subjects['correct']),
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py:74]: bool?

File: brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+                                      'target_present': ('presentation', all_subjects['target_present']),
+                                      'distractor_nums': ('presentation', all_subjects['distractor_nums']),
+                                      'block': ('presentation', all_subjects['block']),
```
Comment (mschrimpf) [inline brainscore_vision/benchmarks/ferguson2024/data_packaging/data_packaging.py:77]: int?

**Review by mschrimpf — APPROVED** [review]
lgtm once the `brainscore_vision/__init__.py` change is undone

**Comment by mike-ferguson** [comment]
Closed in favor of #894
